File: Habitica/src/main/java/com/habitrpg/android/habitica/models/invitations/GuildInvite.java
package com.habitrpg.android.habitica.models.invitations;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class GuildInvite extends RealmObject {

    Invitations invitations;
    private String inviter;
    private String name;
    @PrimaryKey
    private String id;

    private Boolean publicGuild;

    /**
     * @return The inviter
     */
    public String getInviter() {
        return inviter;
    }

    /**
     * @param inviter The inviter
     */
    public void setInviter(String inviter) {
        this.inviter = inviter;
    }

    /**
     * @return The name
     */
    public String getName() {
        return name;
    }

    /**
     * @param name The name
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * @return The id
     */
    public String getId() {
        return id;
    }

    /**
     * @param id The id
     */
    public void setId(String id) {
        this.id = id;
    }

    public Boolean getPublicGuild() {
        return publicGuild;
    }

    public void setPublicGuild(Boolean publicGuild) {
        this.publicGuild = publicGuild;
    }
}

File: shared/src/androidMain/kotlin/com/habitrpg/shared/habitica/models/invitations/GuildInvite.kt
package com.habitrpg.shared.habitica.models.invitations

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

actual open class GuildInvite : RealmObject() {

    @PrimaryKey
    actual var id: String? = null

    internal actual var invitations: Invitations? = null

    actual var inviter: String? = null

    actual var name: String? = null

    actual var publicGuild: Boolean? = null
}
